src/App.js

```diff
@@ -1,16 +1,55 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog';
+
 
 const App = () => {
+  const copyChatMessages = chatMessages.map((chat) => {
+    return {...chat};
+  });
+
+  const [entries, setEntries] = useState(copyChatMessages);
+  const [numLikes, setNumLikes] = useState(0);
+
+  let likeSum = 0
+
+  const likeCounter = (newChatList) => {
+    for (const chat of newChatList) {
+      if (chat.liked === true) {
+        likeSum += 1;
+      };
+    };
+    setNumLikes(likeSum);
```

Comment:
Could you make this this an O(1) operation rather than an O(n)?  Every time updateLiked is called, one is effectively added or subtracted from the number of likes.